### PR TITLE
[3.0][cdc-common] Add ConText for Source/Sink initialization.

### PR DESCRIPTION
**Pull request by joyCurry30**
Refer to #2666
We should add 'ConText' with options for Source/Sink initialization.

### CODE REVIEW

**Comment by joyCurry30** [comment]
@lvyanquan @GOODBOY008 @PatrickRen PTAL.

**Comment by lvyanquan** [comment]
Pass Context to DataSinkFactory/[DataSourceFactory](https://github.com/ververica/flink-cdc-connectors/blob/112b8357437c8db70f5a59d5bcf6727ccae48c99/flink-cdc-common/src/main/java/com/ververica/cdc/common/factories/DataSourceFactory.java), like [DynamicTableSourceFactory](https://github.com/apache/flink/blob/648b6bac95232c8498b392ac01e5089777553c77/flink-table/flink-table-common/src/main/java/org/apache/flink/table/factories/DynamicTableSourceFactory.java).

File: flink-cdc-common/src/main/java/com/ververica/cdc/common/factories/FactoryHelper.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+package com.ververica.cdc.common.factories;
+
+import org.apache.flink.annotation.PublicEvolving;
+
+import java.util.Map;
+
+/** A helper for working with {@link Factory}. */
+@PublicEvolving
+public class FactoryHelper {
```
Comment (lvyanquan) [inline flink-cdc-common/src/main/java/com/ververica/cdc/common/factories/FactoryHelper.java:25]: Not sure this class is necessary, @GOODBOY008 What do you think?

**Comment by lvyanquan** [comment]
[getConfiguration](https://github.com/apache/flink/blob/master/flink-table/flink-table-common/src/main/java/org/apache/flink/table/factories/DynamicTableFactory.java#L162C9-L162C9) is also necessary to get ConfigOption, so you may need to wait https://github.com/ververica/flink-cdc-connectors/issues/2668 to be done.

File: flink-cdc-common/src/main/java/com/ververica/cdc/common/factories/Factory.java [hunk 2]
```diff
@@ -19,7 +19,10 @@
 import org.apache.flink.configuration.ConfigOption;
```
Comment (lvyanquan) [inline flink-cdc-common/src/main/java/com/ververica/cdc/common/factories/Factory.java:19]: Can you also update this import to com.ververica.cdc.common.configuration.ConfigOption?

Comment (joyCurry30) [reply flink-cdc-common/src/main/java/com/ververica/cdc/common/factories/Factory.java:19]: I will fix this in another PR.

**Review by lvyanquan — APPROVED** [review]
LGTM.

File: flink-cdc-connect/flink-cdc-source-connectors/pom.xml [hunk 3]
```diff
@@ -50,4 +50,18 @@ under the License.
...
+        <dependency>
+            <groupId>com.ververica</groupId>
+            <artifactId>flink-cdc-common</artifactId>
+            <version>${revision}</version>
+        </dependency>
+
+        <dependency>
+            <groupId>com.ververica</groupId>
+            <artifactId>flink-cdc-runtime</artifactId>
+            <version>${revision}</version>
```
Comment (whhe) [inline flink-cdc-connect/flink-cdc-source-connectors/pom.xml:63]: Should use ${project.version} in dependencies.

Comment (joyCurry30) [reply flink-cdc-connect/flink-cdc-source-connectors/pom.xml]: Done. Thank you for advice.

**Pull request merged**